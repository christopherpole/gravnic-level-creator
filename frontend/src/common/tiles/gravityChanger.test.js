import React from 'react';
import { MOVE_DOWN } from 'gravnic-game';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import GravityChangerTile from './gravityChanger';

configure({ adapter: new Adapter() });

describe('<GravityChangerTile />', () => {
  let props;

  beforeEach(() => {
    props = {
      direction: MOVE_DOWN,
    };
  });

  it('Renders without exploding', () => {
    const gravityChanger = shallow(<GravityChangerTile {...props} />);

    expect(gravityChanger).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const gravityChanger = shallow(<GravityChangerTile {...props} />);

    expect(toJson(gravityChanger)).toMatchSnapshot();
  });
});
