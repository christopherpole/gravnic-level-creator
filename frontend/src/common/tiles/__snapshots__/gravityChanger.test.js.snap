// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GravityChangerTile /> Matches the current snapshot 1`] = `
<styled.svg
  height="100%"
  rotation={180}
  version="1.1"
  viewBox="0 0 100 100"
  width="100%"
>
  <defs />
  <g
    fill="none"
    fillRule="evenodd"
    stroke="none"
    strokeWidth="0"
  >
    <g>
      <rect
        fill="#FFFFFF"
        height="100"
        width="100"
        x="0"
        y="0"
      />
      <polygon
        fill="#D8D8D8"
        points="50 25 75 75 25 75"
        stroke="#979797"
        strokeWidth="2"
      />
    </g>
  </g>
</styled.svg>
`;
